### PR TITLE
47612 fix primitive array

### PR DESCRIPTION
**Pull request by mrix**
Fixed processing JSON definitions for field that contains array of primitives:
```json
{
  "name": "arrayOfStrings.0",
  "type": "string"
}
```

- [x] update definition parsing
- [x] update JSON definition schema
- [x] update Twig templates:
 - [x] update Document class template
 - [x] update Doctrine mapping template
 - [x] update Serializer mapping template
 - [x] update Validation constraints template
- [x] update form building:
 - [x] add Doctrine type for `array<object>`
 - [x] add Form builder for new types
- [x] functional test for array fields:
 - [x] service definition for new test case
 - [x] functional test case

### CODE REVIEW

**Comment by narcoticfresh** [comment]
@mrix first off, thanks for this feature.. primitive array support is long overdue ;-)

one important thing: in libgraviton/GravitonTestServicesBundle#16 you write not to merge the test fixtures before this PR.. how should we test that before merging? ;-)

so, what we shall do: we merge libgraviton/GravitonTestServicesBundle#16 into develop (no tagging). then, in this PR, you set the dependency *temporarly* to `graviton/test-services-bundle: dev-develop`.. then, we'll let the test run.. 

if all is fine, we tag libgraviton/GravitonTestServicesBundle and you bump to the new release in this PR.. 

as all dependencies from others refer to tagged releases, the only thing this would block is that they are not able to bump the new libgraviton/GravitonTestServicesBundle tag unless this PR is merged.. 

but first, please get your TODO above done (schema)..

**Comment by mrix** [comment]
Hi @narcoticfresh. 
I've finished working on this feature. Please merge https://github.com/libgraviton/GravitonTestServicesBundle/pull/16 into `develop` branch.

**Comment by hairmare** [comment]
libgraviton/GravitonTestServicesBundle#16 was merged and tagged into a release.

**Comment by mrix** [comment]
@hairmare thanks. I have updated `graviton/test-services-bundle` to v0.10.0. All checks have passed.

File: src/Graviton/SchemaBundle/SchemaUtils.php [hunk 1]
```diff
@@ -231,6 +240,28 @@ function ($record) {
                     }
                 }
                 $property->setRefCollection($urls);
+            } elseif ($meta->getTypeOfField($field) === 'collection') {
+                $itemSchema = new Schema();
+                $property->setType('array');
+                $itemSchema->setType($this->getCollectionItemType($meta->name, $field));
```
Comment (hairmare) [inline src/Graviton/SchemaBundle/SchemaUtils.php]: How can these be properly documented with a title and description?

Comment (mrix) [inline src/Graviton/SchemaBundle/SchemaUtils.php]: We get the following JSON schema:
```json
"intarray": {
    "title": "Intarray title",
    "description": "Intarray description",
    "type": "array",
    "items": {
        "type": "integer"
    }
}
```

for the field definition:
```json
{
    "name": "intarray.0",
    "title": "Intarray title",
    "description": "Intarray description",
    "type": "int"
}
```

Of course I can add `title` field to the `items` schema but I dont know what should be there.

Comment (hairmare) [inline src/Graviton/SchemaBundle/SchemaUtils.php]: Thanks for the example, this does make sense for most cases. We'll have to check if this doesn't look weird in graviphoton, so we might have to add something.

Comment (mrix) [inline src/Graviton/SchemaBundle/SchemaUtils.php]: So should I add anything like `"title": "<arrayfieldtitle> item"` to the item schema now?

Comment (hairmare) [inline src/Graviton/SchemaBundle/SchemaUtils.php]: I asked the graviphoton peeps for feedback.

FWIW this is how your example currently gets rendered:

screen shot 2015-10-05 at 12 56 10

Comment (hairmare) [inline src/Graviton/SchemaBundle/SchemaUtils.php]: Graviphoton thinks this can stay like this for the time being.

File: composer.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
         "libgraviton/codesniffer": "~1.3",
         "lapistano/proxy-object": "dev-master",
         "johnkary/phpunit-speedtrap": "~1.0",
-        "graviton/test-services-bundle": "^0"
+        "graviton/test-services-bundle": "v0.10.0"
```
Comment (hairmare) [inline composer.json]: Please allow versions larger than the specified one as well.

**Pull request merged**